### PR TITLE
feat: add prefix option to all created resources DA

### PR DESCRIPTION
**Pull request by iamar7**
### Description

Issue: https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/issues/94

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by akocbek** [comment]
@iamar7 please check how we add it prefix in https://github.com/terraform-ibm-modules/terraform-ibm-observability-da/pull/33

- for better readability, you can introduce local variable
- specifically define that prefix is optional
- add prefix option to other  input variables description

**Review by alex-reiff — DISMISSED** [review]

File: tests/pr_test.go [hunk 1]
```diff
@@ -128,8 +128,10 @@ func TestRunDASolutionSchematics(t *testing.T) {
 
 	options.TerraformVars = []testschematic.TestSchematicTerraformVar{
 		{Name: "ibmcloud_api_key", Value: options.RequiredEnvironmentVars["TF_VAR_ibmcloud_api_key"], DataType: "string", Secure: true},
+		{Name: "prefix", Value: options.Prefix, DataType: "string"},
 		{Name: "region", Value: options.Region, DataType: "string"},
-		{Name: "resource_group_name", Value: options.Prefix, DataType: "string"},
+		{Name: "existing_resource_group", Value: true, DataType: "string"},
```
Comment (ocofaigh) [inline tests/pr_test.go:133]: remove this - we want the test to test creating a new resource group using prefix

File: tests/pr_test.go [hunk 2]
```diff
@@ -128,8 +128,10 @@ func TestRunDASolutionSchematics(t *testing.T) {
...
+		{Name: "resource_group_name", Value: options.ResourceGroup, DataType: "string"},
```
Comment (ocofaigh) [inline tests/pr_test.go:134]: change this back to `options.Prefix`

File: solutions/standard/variables.tf [hunk 3]
```diff
@@ -118,13 +124,13 @@ variable "kms_endpoint_type" {
...
   default     = "sm-cos-key-ring"
-  description = "The name to give to the new KMS key ring that will be used to store the KMS key to enable advanced, customer-managed encryption for your Secrets Manager secrets. Only required if not supplying an existing KMS root key CRN."
+  description = "The name to give the Key Ring which will be created for the Secrets Manager COS bucket Key. Not used if supplying an existing Key. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'"
 }
 
 variable "kms_key_name" {
   type        = string
   default     = "sm-cos-key"
-  description = "The name to give to the new KMS root key that will be used to enable advanced, customer-managed encryption for your Secrets Manager secrets. Only required if not supplying an existing KMS root key CRN."
+  description = "The name to give the Key which will be created for the Secrets Manager COS bucket. Not used if supplying an existing Key. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'"
```
Comment (ocofaigh) [inline solutions/standard/variables.tf:133]: why did you change the description to say COS bucket? There is no COS bucket here

File: solutions/standard/variables.tf [hunk 4]
```diff
@@ -118,13 +124,13 @@ variable "kms_endpoint_type" {
 variable "kms_key_ring_name" {
   type        = string
   default     = "sm-cos-key-ring"
-  description = "The name to give to the new KMS key ring that will be used to store the KMS key to enable advanced, customer-managed encryption for your Secrets Manager secrets. Only required if not supplying an existing KMS root key CRN."
+  description = "The name to give the Key Ring which will be created for the Secrets Manager COS bucket Key. Not used if supplying an existing Key. If prefix input variable is passed then it will get prefixed infront of the value in the format of '<prefix>-value'"
```
Comment (ocofaigh) [inline solutions/standard/variables.tf:127]: why did you change the description to say COS bucket? There is no COS bucket here

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: tests/pr_test.go (hunk 1 above)
Comment (iamar7) [reply tests/pr_test.go:133]: Shall I remove this assignment as well from the `TestRunDASolutionSchematics` -
`ResourceGroup:          resourceGroup,`

File: solutions/standard/variables.tf (hunk 3 above)
Comment (iamar7) [reply solutions/standard/variables.tf:133]: As can be seen from the main [variable.tf](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/blob/60accca4ed5a83b4afd45f002ec831ae711be226/solutions/standard/variables.tf#L127) the description had COS bucket in it.

Comment (iamar7) [reply solutions/standard/variables.tf:133]: The earlier variable description had COS bucket in it but it was modified in this [PR](https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/commit/4593f02bf30c3b59b820a650b360bff813bacbad) which may not have been taken care while merging the main branch.

File: tests/pr_test.go (hunk 1 above)
Comment (ocofaigh) [reply tests/pr_test.go:133]: No this indicates the resource group that the schematics workspace will be created in, so its needed for that

**Review by ocofaigh — APPROVED** [review]
